pkg/sql/parser/sql.y

```diff
@@ -8124,6 +8124,10 @@ sequence_option_elem:
   AS typename                  {
                                   // Valid option values must be integer types (ex. int2, bigint)
                                   parsedType := $2.colType()
+                                  if parsedType == nil {
+                                      sqllex.Error(fmt.Sprintf("invalid type name: %s", $2.val))
```

Comment:
The error in Postgres looks like:

```
marcus=# create sequence s1 as abc;
ERROR:  42704: type "abc" does not exist
LINE 1: create sequence s1 as abc;
```

Can you try to make the error message read similarly, like "type %q does not exist", and make the error code match, [`42704`](https://github.com/cockroachdb/cockroach/blob/50df58d2f415a44ef15adc1482f21a18179964f3/pkg/sql/pgwire/pgcode/codes.go#L222)?